stock_picking_invoice_link/models/stock_move.py

```diff
@@ -3,15 +3,27 @@
 # Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
-from odoo import fields, models
+from odoo import _, fields, models
+from odoo.exceptions import UserError
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
-    invoice_line_id = fields.Many2one(
+    invoice_line_ids = fields.Many2many(
         comodel_name='account.invoice.line',
+        relation='stock_move_invoice_line_rel',
+        column1='move_id',
+        column2='invoice_line_id',
         string='Invoice Line',
         copy=False,
         readonly=True,
     )
+
+    def write(self, vals):
+        if 'product_uom_qty' not in vals:
+            return super().write(vals)
+        for move in self:
+            if move.state == 'done' and move.invoice_line_ids:
```

Comment:
This is difficult to be read, having also 2 calls to super. Please use 3 levels of indentation this time.